File: src-tauri/src/models/usage.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LlmUsageRecord {
    pub id: String,
    pub agent_id: String,
    pub model_config_id: String,
    pub session_id: Option<String>,
    pub trigger_type: String,
    pub call_round: i32,
    pub prompt_tokens: i32,
    pub completion_tokens: i32,
    pub total_tokens: i32,
    pub message_id: Option<String>,
    pub created_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UsageOverview {
    pub total_calls: i64,
    pub total_prompt_tokens: i64,
    pub total_completion_tokens: i64,
    pub total_tokens: i64,
    pub daily_trend: Vec<DailyTrend>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DailyTrend {
    pub date: String,
    pub calls: i64,
    pub tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ModelUsageItem {
    pub model_config_id: String,
    pub model_name: String,
    pub provider: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentUsageItem {
    pub agent_id: String,
    pub agent_name: String,
    pub avatar_path: Option<String>,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AgentModelUsageItem {
    pub model_config_id: String,
    pub model_name: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ModelAgentUsageItem {
    pub agent_id: String,
    pub agent_name: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionUsageItem {
    pub session_id: String,
    pub session_name: String,
    pub session_type: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionAgentUsageItem {
    pub agent_id: String,
    pub agent_name: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionModelUsageItem {
    pub model_config_id: String,
    pub model_name: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SessionAgentModelUsageItem {
    pub agent_id: String,
    pub agent_name: String,
    pub model_config_id: String,
    pub model_name: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TriggerUsageItem {
    pub trigger_type: String,
    pub calls: i64,
    pub prompt_tokens: i64,
    pub completion_tokens: i64,
    pub total_tokens: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UsageFilters {
    pub agent_id: Option<String>,
    pub model_config_id: Option<String>,
    pub session_id: Option<String>,
    pub trigger_type: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UsageRecordDetail {
    pub id: String,
    pub agent_name: String,
    pub model_name: String,
    pub session_name: Option<String>,
    pub trigger_type: String,
    pub call_round: i32,
    pub prompt_tokens: i32,
    pub completion_tokens: i32,
    pub total_tokens: i32,
    pub created_at: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PaginatedUsageRecords {
    pub records: Vec<UsageRecordDetail>,
    pub total: i64,
    pub page: i32,
    pub page_size: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TimeRange {
    pub start_time: i64,
    pub end_time: i64,
}

File: src/lib/types/usage.ts
export interface UsageOverview {
    total_calls: number;
    total_prompt_tokens: number;
    total_completion_tokens: number;
    total_tokens: number;
    daily_trend: DailyTrend[];
}

export interface DailyTrend {
    date: string;
    calls: number;
    tokens: number;
}

export interface ModelUsageItem {
    model_config_id: string;
    model_name: string;
    provider: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface AgentUsageItem {
    agent_id: string;
    agent_name: string;
    avatar_path: string | null;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface AgentModelUsageItem {
    model_config_id: string;
    model_name: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface ModelAgentUsageItem {
    agent_id: string;
    agent_name: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface SessionUsageItem {
    session_id: string;
    session_name: string;
    session_type: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface SessionAgentUsageItem {
    agent_id: string;
    agent_name: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface SessionModelUsageItem {
    model_config_id: string;
    model_name: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface SessionAgentModelUsageItem {
    agent_id: string;
    agent_name: string;
    model_config_id: string;
    model_name: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface TriggerUsageItem {
    trigger_type: string;
    calls: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
}

export interface UsageFilters {
    agent_id?: string;
    model_config_id?: string;
    session_id?: string;
    trigger_type?: string;
}

export interface UsageRecordDetail {
    id: string;
    agent_name: string;
    model_name: string;
    session_name: string | null;
    trigger_type: string;
    call_round: number;
    prompt_tokens: number;
    completion_tokens: number;
    total_tokens: number;
    created_at: number;
}

export interface PaginatedUsageRecords {
    records: UsageRecordDetail[];
    total: number;
    page: number;
    page_size: number;
}

export type TimeRange = 'today' | 'last_7_days' | 'last_30_days' | 'this_month' | 'all';

export const TRIGGER_TYPE_LABELS: Record<string, string> = {
    user_message: '用户消息触发',
    background_scan: '后台扫描',
    timer: '定时任务',
    proactive: '主动会话',
    persona_generation: '人设生成',
    tts_translate: 'TTS 翻译',
};
